src/data/nat/prime.lean

```diff
@@ -1102,41 +1102,63 @@ end nat
 namespace nat
 
 /-- The only prime divisor of positive prime power `p^k` is `p` itself -/
-lemma prime_pow_prime_divisor {p k : ℕ} (hk : 0 < k) (hp: prime p) :
+lemma prime_pow_prime_divisor {p k : ℕ} (hk : k ≠ 0) (hp : prime p) :
   (p^k).factors.to_finset = {p} :=
-by rw [hp.factors_pow, list.to_finset_repeat_of_ne_zero hk.ne']
+by rw [hp.factors_pow, list.to_finset_repeat_of_ne_zero hk]
```

Comment:
If you move this down to come below `pow_factors_to_finset` then I think you can reduce the proof to
```
by simp [pow_factors_to_finset p hk, factors_prime hp]
```